enhancements/network/egressip_for_ovn_kubernetes.md

```diff
@@ -0,0 +1,134 @@
+
+This enhancement proposal focuses on simplifying this process as well as adding a `podSelector` to the `EgressIP` object, allowing a user to specify a subset of pods within the namespace which should have an egress IP defined. 
+
+If we resume what openshift-sdn concretely achieves by its two-step process, we have the following: 
+
+1.1. It allows the cluster admin to have a direct say on what nodes act as egress nodes.
+1.2. The final assignment of egress IPs is still a manual process, and even though the re-distribution of them, warrented by the HA mechanism, is automatic (in case of a node failure), the newly distributed egress IP will remain the same. 
+
+All in all, the feature is quite complex as it introduces a lot of notions which obfuscate these two final functionalities/goals.  
+
+## Motivation
+
+The motivation behind this requirement comes from partially from product management (assigning a `podSelector`  and essentially allowing cluster admins the freedom to control cluster egress traffic on a lower granular level), and partially from us in the networking team (simplifying the usability for cluster admins and our maintenance of the code base which implements it).   
+
+### Goals & Requirements
+
+2.1. The egress IP functionality for OVN-Kubernetes needs to have feature parity with openshift-sdn, according to 1.1. and 1.2.
+2.2. The egress IP functionality for OVN-Kubernetes should allow pod selection within a namespace when defining the IP of egress traffic. 
+2.3. The egress IP functionality for OVN-Kubernetes should support single-stack IPv4/IPv6 as well as dual-stack, in line with global OpenShift requirements for all networking functionality for release 4.6.
+
+This proposal details how OVN-Kubernetes goes about addressing these goals in the section [Proposal](## Proposal)
+ 
+### Constraints
+
+3.1. No pod can match multiple `EgressIP` objects, leading to multiple egress IP assignments at the same time. 
+3.2. Any egress IP can only be assigned to one node, and one node only, at any given moment in time. 
+3.3. An `EgressIP` object can not claim multiple egress IPs for the same node. 
+
+This proposal details how OVN-Kubernetes goes about addressing these constraints in the section [Proposal](## Proposal)
+ 
+### Non-Goals
+
+4.1. Making an EgressIP object cluster-wide, it will still be namespaced
+
+## Proposal
+
+We propose to make the assignment of egress IPs simpler by defining the following sequence of events for a user wishing to define an egress IP for its cluster.
+
+As to satisfy 1.1. we propose that cluster admins tag the nodes they wish to use as egress nodes by labelling them with `node-role.kubernetes.io/egress-assignable: ""`. In case an `EgressIP` object is created and no nodes are tagged as such, all nodes are considered for egress IP assignment. To rectify that: a cluster admin would have to; delete the previously created `EgressIP` object, label the nodes needed, and re-create the previous object. 
```

Comment:
labels for ovn-kubernetes-specific features should be in the `k8s.ovn.org` namespace.